app/assets/javascripts/discourse/app/lib/plugin-api.js

```diff
@@ -940,6 +943,28 @@ class PluginApi {
     extraConnectorClass(`${outletName}/${connectorName}`, klass);
   }
 
+  /**
+   * Register a component to be rendered in a particular outlet.
```

Comment:
We need to update the [CHANGELOG](https://github.com/discourse/discourse/blob/main/docs/CHANGELOG-JAVASCRIPT-PLUGIN-API.md) and also do a version bump as well.